### PR TITLE
Add ability to selectively claim joint interfaces

### PR DESCRIPTION
**Pull request by wxmerkt**
This PR adds the ability to selectively claim joints and thus allows us to have multiple controllers run at the same time, e.g. have IHMC's controller run the effort joints with their whole body controller and us run separate controllers for neck and finger/forearm joints. 

Tested on Val. Sequence to get it to work:
1. start controller manager
2. start this controller using the NeckControl.launch
3. start the IHMC controller (either through command line or through ui_builder)

### CODE REVIEW

**Comment by mauricefallon** [comment]
This badly needs to be functionalised and perhaps the lcm i/o broken out into a separate cpp for readability. The functions inside the namespace shouldn't be indented.

I'd like to do away with having the header in the include directory. We don't use that approach in the rest of the code base

Probably should review this as is and make these changes afterwards

File: src/LCM2ROSControl.cpp [hunk 1]
```diff
@@ -41,7 +41,18 @@ namespace valkyrie_translator
           return false;
         }
         handler_ = std::shared_ptr<LCM2ROSControl_LCMHandler>(new LCM2ROSControl_LCMHandler(*this));
-        
+
+        // Check which joints we have been assigned to
+        // If we have joints assigned to just us, claim those, otherwise claim all
+        std::vector<std::string> joint_names_;
+        if (!controller_nh.getParam("joints", joint_names_))
+          ROS_INFO_STREAM("Could not get assigned list of joints, will resume to claim all");
```
Comment (mauricefallon) [inline src/LCM2ROSControl.cpp:51]: I don't like splitting the config between ROS params and lcm-config and director-json (and possibly also drake config).
this is the kind of decentralisation that I abhor in ROS.

But for now we will use this.

File: src/LCM2ROSControl.cpp [hunk 2]
```diff
@@ -204,43 +272,79 @@ namespace valkyrie_translator
...
       // push out the measurements for all imus we see advertised
+      int i;
       for (auto iter = imuSensorHandles.begin(); iter != imuSensorHandles.end(); iter ++){
         bot_core::ins_t lcm_imu_msg;
         //lcm_imu_msg.utime = utime;
         std::ostringstream imuchannel;
         imuchannel << "VAL_IMU_" << iter->first;
         lcm_imu_msg.utime = utime;
-        for (i=0; i<3; i++){
+        for (int i=0; i<3; i++){
```
Comment (mauricefallon) [inline src/LCM2ROSControl.cpp]: Iterator 0:3 for quaternion seems to be a bug ... missing the 4th element?

Comment (wxmerkt) [inline src/LCM2ROSControl.cpp]: Good catch - probably a bug, need to check this (in sim we don't get IMU data and on the robot we haven't checked it so far).

Comment (mauricefallon) [inline src/LCM2ROSControl.cpp]: Just looked at the raw file. The 4th element was being done manually after the loop. Please redo that as that without using this loop. Is a pretty sloppy way of avoiding a few lines.

File: src/LCM2ROSControl.cpp (hunk 1 above)
Comment (wxmerkt) [inline src/LCM2ROSControl.cpp:51]: Agreed. However for the purpose of a single controller only running on the robot without other dependencies this the preferred and recommended approach. The getParam is read from the yaml file where we specify whether to assign specific joints (e.g. the neck) or whether to grab all available joints (in the case of no joints specified in the parameter). LCM or director json config isn't compatible with ROS Control per se at the moment, but we can look into making it

File: src/LCM2ROSControl.cpp (hunk 2 above)
Comment (wxmerkt) [inline src/LCM2ROSControl.cpp]: Agreed, I didn't revise/clean up bits that weren't needed to be touched yet. Will do in the next PR

**Pull request merged**